### ISSUE TITLE
User mapping for Merge Requests

### ISSUE THREAD

**Issue by mike-potter** [piceaTech/node-gitlab-2-github#104]
I have filled out the user mapping table between GitLab and GitHub usernames.  However, when I migrate the MRs->PRs the owner of the PR in Github is always my own user and not the user from GitLab.  Is there a way to fix that?

**Comment by mdbenito** [comment]
Not really. The GH api sets the owner to the one making the request. The only way is to make requests as the different users (with different tokens) or to use an impersonation token, as available on GH Enterprise. This script does not allow for that (yet)

**Comment by spruce** [comment]
@mike-potter with the new setting `useIssueImportAPI` it may be possible. @mdbenito did you test that?

**Comment by mdbenito** [comment]
Sadly, the issue import API [does not allow](https://gist.github.com/jonmagic/5282384165e0f86ef105#supported-issue-and-comment-fields) setting the creator of an issue. The only solutions I know of are the two I listed (and I'm not sure about how the impersonation token would work)

**Comment by XabierGoros** [comment]
Have you guys checked [this](https://github.com/piceaTech/node-gitlab-2-github/compare/master...deckweiss-gmbh:node-gitlab-2-github:master)?

**Comment by mdbenito** [comment]
That would be a very useful contribution! Can we expect a PR? 😃

**Comment by XabierGoros** [comment]
Not my changes, all credits are for @valentinschabschneide from his repo. He should definitely be contacted. I personally tested those changes with great success.

**Comment by valentinschabschneider** [comment]
@XabierGoros totally forgot about that repo 😅

Cool to see that @pelme is now implementing it properly 🎉

**Comment by pelme** [comment]
@valentinschabschneider I assume you already migrated your repos but a quick code review from you would be very helpful.🙏 

There were a handful of merge conflicts to deal with, I hope I didn’t screw anything up 🙂
